.travis.yml

```diff
@@ -1,48 +1,83 @@
 sudo: true
-language: haskell
+language: generic
 
 git:
   depth: 5
 
 cache:
   directories:
+  - "$HOME/.cabal"
   - "$HOME/.stack"
-  - "$HOME/build/kowainik/issue-wanted/.stack-work"
+  - "$TRAVIS_BUILD_DIR/.stack-work"
 
 matrix:
   include:
-
   - ghc: 8.2.2
-    env: GHCVER='8.2.2' STACK_YAML="$HOME/build/kowainik/issue-wanted/stack.yaml"
+    env: GHCVER='8.2.2' CABALVER='head'
+    os: linux
+    addons:
+      apt:
+        sources:
+        - hvr-ghc
+        packages:
+        - ghc-8.2.2
+        - cabal-install-head
 
-addons:
-  apt:
-    sources:
-      - sourceline: 'ppa:hvr/ghc'
-    packages:
-      - libgmp-dev
+  - ghc: 8.4.3
+    env: GHCVER='8.4.3' CABALVER='head'
+    os: linux
+    addons:
+      apt:
+        sources:
+        - hvr-ghc
+        packages:
+        - ghc-8.4.3
+        - cabal-install-head
 
-before_install:
-  - mkdir -p ~/.local/bin
-  - export PATH="$HOME/.local/bin:$PATH"
-  - travis_retry curl -L 'https://www.stackage.org/stack/linux-x86_64' | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
-  - stack --version
+  - ghc: 8.2.2
+    env: GHCVER='8.2.2' STACK_YAML="$TRAVIS_BUILD_DIR/stack-$GHCVER.yaml"
+    os: linux
+    addons:
+      apt:
+        packages:
+        - libgmp-dev
 
+  - ghc: 8.4.3
+    env: GHCVER='8.4.3' STACK_YAML="$TRAVIS_BUILD_DIR/stack.yaml"
+    os: linux
+    addons:
+      apt:
+        packages:
+        - libgmp-dev
 
 install:
-  - travis_wait 30 stack setup --no-terminal
-  - stack ghc -- --version
-
-  - travis_wait 40 stack build --only-dependencies --no-terminal
-  - travis_wait 40 stack build --test --bench --haddock --no-run-tests --no-run-benchmarks --no-haddock-deps --no-terminal
-
+  - |
+    if [ -z "$STACK_YAML" ]; then
+      export PATH="/opt/ghc/$GHCVER/bin:/opt/cabal/$CABALVER/bin:$PATH"
+      echo $PATH
+      cabal new-update
+      cabal new-build --enable-tests --enable-benchmarks
+    else
+      mkdir -p ~/.local/bin
+      export PATH="$HOME/.local/bin:$PATH"
+      travis_retry curl -L 'https://www.stackage.org/stack/linux-x86_64' | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
+      stack --version
+      stack setup --no-terminal --install-cabal 2.2.0.1
+      stack ghc -- --version
+      stack build --only-dependencies --no-terminal
+    fi
 script:
-  - travis_wait 40 stack build --test --no-terminal
+  - |
+    if [ -z "$STACK_YAML" ]; then
+       cabal new-test
+    else
+      stack build --test --no-terminal
+    fi
 
 notifications:
   email: false
   slack:
     rooms:
-      - kowainik:k3hrOZtFA7ViWPFCLrSokLcU
+      - kowainik:n60CMbB0kfJHRmYKIPmjf7Tp
```

Comment:
We can keep the previous room number (it's #issue-wanted-apps in our Slack, at least it should be so)